src/hooks/useFormField.js

```diff
@@ -1,27 +1,52 @@
 import { useState } from "react";
-/*
-A custom hook controlling form field logic.
-It could be extended with:
-    - validation support (a function with validation rules could be passed as an argument)
-    - error message states 
-*/
+import { validateMultipleRules } from "../utils/ValidateRule/ValidateRule";
 
-export default function useFormField(initialState) {
-  const [currentState, setState] = useState(initialState);
+/**
+ * Hook to handle fields and validate them.
+ *
+ * @param {string} initialState - The intial value for the input field.
+ * @param {InputValidator} InputValidator - ValidatorRule that is used to validate the field.
+ *
+ */
 
-  function handleInputChange(e) {
-    switch (e.target.type) {
-      case "checkbox":
-        setState(e.target.checked);
-        break;
-      default:
-        setState(e.target.value);
-    }
-  }
+export default function useFormField(initialState, InputValidator) {
+  const NO_INPUT_VALIDATION = "Input validation is undefinied.";
+
+  const [currentState, setState] = useState(initialState);
+  const [inputValidation, setInputValidation] = useState(InputValidator);
+  const [isInputValid, setIsInputValid] = useState(true);
+  const [errorMessage, setErrorMessage] = useState(NO_INPUT_VALIDATION);
 
   function resetInputState() {
     setState("");
   }
 
-  return [currentState, handleInputChange, resetInputState];
+  function validateInput() {
+    if (inputValidation === undefined) {
+      setIsInputValid(false);
```

Comment:
I'm not sure that I understand this: if the inputValidator is not defined, how can we say that the input is not valid? I guess I don't understand something.

Reply:
I think this is because I am expecting if you are calling the validateInput because you have set an inputValidator, otherwise you wouldn't call the validateInput. So if it's undefined it's more of an error that the inputValidator should have been defined in the first place. That was my thought process. Since if you are not validating anything you could instead just use a standard UseState from React.


I can change it, so we could still use this hook without a validator, but then it seems to me that it's missing the point of the class. I guess I could also just throw an exception to make it clear that it expects a validator to be passed?

Reply:
I will allow the hook to be used without an InputValidator and in that case it will just return true.